web/components/badges/badges-dialog.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from 'react'
+import { Col } from '../layout/col'
+import { Row } from '../layout/row'
+import {
+  LightningBoltIcon,
+  PresentationChartLineIcon,
+} from '@heroicons/react/solid'
+import { Badge } from './badge'
+import { Modal } from '../layout/modal'
+import { Title } from '../title'
+import { User } from 'web/lib/firebase/users'
+
+export function BadgesDialog(props: { user: User }) {
+  const { user } = props
+  const [open, setOpen] = useState(false)
+
+  const topCreatorBadge = () => (
+    <Badge
+      icon={<PresentationChartLineIcon className="h-5 w-5" />}
+      label={'#' + user.badges.weeklyCreatorRank}
+    />
+  )
+
+  const topTraderBadge = () => (
+    <Badge
+      icon={<LightningBoltIcon className="h-5 w-5" />}
+      label={'#' + user.badges.weeklyTraderRank}
+    />
+  )
```

Comment:
These don't need to be functions

Reply:
If we end up adding dozens of different types of badges, it's nicer to have them as functions rather than cluttering the main return statement. But can rm for now.